storage/src/backend/registry.rs

```diff
@@ -394,18 +468,30 @@ impl RegistryReader {
         offset: u64,
         allow_retry: bool,
     ) -> RegistryResult<usize> {
+        let state = self.get_state();
         let url = format!("/blobs/sha256:{}", self.blob_id);
-        let url = self
-            .state
-            .url(url.as_str(), &[])
-            .map_err(RegistryError::Url)?;
+        let url = state.url(url.as_str(), &[]).map_err(RegistryError::Url)?;
         let mut headers = HeaderMap::new();
         let end_at = offset + buf.len() as u64 - 1;
         let range = format!("bytes={}-{}", offset, end_at);
         headers.insert("Range", range.parse().unwrap());
 
+        if !self.is_registry() {
+            let mirror_index = self.mirrors.index.load(Ordering::Relaxed) as usize;
+            let mirror = self.mirrors.states[mirror_index].clone();
+            headers.insert(
```

Comment:
Header should be gathered from backend json configuration and directly transfer to HTTP requests header lines. Nydusd does not care about the heard Key name.